fetch-in-react/my-app/src/Users.tsx

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserDetails } from './UserDetails';
+import { UserCard } from './UserCard';
+import './Users.css';
+
+export type User = {
+  id: number;
+  name: string;
+  email: string;
+  company: {
+    name: string;
+  };
+};
+export function Users() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [users, setUsers] = useState<User[]>([]);
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      try {
+        const response = await fetch(
+          'https://jsonplaceholder.typicode.com/users'
+        );
+        if (!response.ok) {
+          throw new Error('response failed');
+        }
+        const data = (await response.json()) as User[];
+        setUsers(data);
+        setIsLoading(false);
+      } catch (err) {
+        setError(err);
+        setIsLoading(false);
```

Comment:
Since you need to run `setIsLoading(false)` in both cases, this is where the `finally` statement comes into play. So you could do

```
try {
  const response = await fetch("https://jsonplaceholder.typicode.com/users");
  if (!response.ok) {
    throw new Error("response failed");
  }
  const data = (await response.json()) as User[];
  setUsers(data);
} catch (err) {
  setError(err);
} finally {
  setIsLoading(false);
}

```